proof_of_stake/src/tests.rs

```diff
@@ -755,27 +758,61 @@ fn test_validator_sets() {
         max_validator_slots: 3,
         ..Default::default()
     };
-    let seed = "seed";
-    let mut address_gen = EstablishedAddressGen::new(seed);
-    let mut gen_validator = || address_gen.generate_address(seed);
+    let addr_seed = "seed";
+    let mut address_gen = EstablishedAddressGen::new(addr_seed);
+    let mut sk_seed = 0;
+    let mut gen_validator = || {
+        let res = (
+            address_gen.generate_address(addr_seed),
+            key::testing::common_sk_from_simple_seed(sk_seed).to_public(),
+        );
+        // bump the sk seed
+        sk_seed += 1;
+        res
+    };
+
+    // A helper to insert a non-genesis validator
+    let insert_validator = |s: &mut TestWlStorage,
+                            addr,
+                            pk: &PublicKey,
+                            stake: token::Amount,
+                            epoch: Epoch| {
+        insert_validator_into_validator_set(
+            s,
+            &params,
+            addr,
+            stake,
+            epoch,
+            params.pipeline_len,
+        )
+        .unwrap();
+
+        update_validator_deltas(s, &params, addr, stake.change(), epoch)
```

Comment:
is it useful at all (for consistency) to update total deltas too? Maybe not necessary for the test.